x-pack/plugins/apm/server/lib/apm_telemetry/collect_data_telemetry/tasks.ts

```diff
@@ -1082,6 +1093,79 @@ export const tasks: TelemetryTask[] = [
   {
     name: 'indices_stats',
     executor: async ({ indices, telemetryClient }) => {
+      const dsRollupDictionary: DataStreamWithRollup =
+        {} as DataStreamWithRollup;
+      const dsWithoutRollupDictionary: DataStreamWithoutRollup =
+        {} as DataStreamWithoutRollup;
+
+      const metricSetsSupportingRollUps: MetricSupportingRollUp[] = [
+        MetricTypes.service_destination,
+        MetricTypes.transaction,
+        MetricTypes.service_summary,
+        MetricTypes.service_transaction,
+        MetricTypes.span_breakdown,
+      ];
+
+      const metricSetsNotSupportingRollUps: MetricNotSupportingRollup[] = [
+        MetricTypes.app,
+      ];
+
+      const rollUpIntervals: MetricRollupIntervals[] = [
+        RollupInterval.OneMinute,
+        RollupInterval.TenMinutes,
+        RollupInterval.SixtyMinutes,
+      ];
+
+      // The API calls must be done in series rather than in parallel due to the nature
+      // of how tasks are executed. We don't want to burden the customers instances
+      // which could directly impact the performance on the UI.
+      const fetchRollupMetrics = async () => {
+        for (const metricSet of metricSetsSupportingRollUps) {
+          for (const bucketSize of rollUpIntervals) {
+            const datastream = `metrics-apm.${metricSet}.${bucketSize}-default*`;
+            const response = await telemetryClient.indicesStats({
+              index: [datastream],
+              expand_wildcards: 'all',
+              filter_path: ['_all', '_shards'],
```

Comment:
This could be narrowed down even further but not sure if it'll result in considerable improvements

Reply:
Given we are already filtering out the actual `indices` field present (which could be huge), the payload is significantly smaller now. I will still narrow them further down as its no loss.

```
filter_path: [
              '_all.primaries.docs',
              '_all.primaries.store',
              '_shards',
            ],
```